include/boost/python/detail/value_is_xxx.hpp

```diff
@@ -9,23 +9,21 @@
 # include <boost/mpl/bool.hpp>
 # include <boost/preprocessor/enum_params.hpp>
 
-
-#  include <boost/type_traits/remove_reference.hpp>
-#  include <boost/type_traits/remove_cv.hpp>
+# include <boost/python/detail/type_traits.hpp>
 #  include <boost/python/detail/is_xxx.hpp>
 
-#  define BOOST_PYTHON_VALUE_IS_XXX_DEF(name, qualified_name, nargs)    \
-template <class X_>                                                     \
-struct value_is_##name                                                  \
-{                                                                       \
-    BOOST_PYTHON_IS_XXX_DEF(name,qualified_name,nargs)                  \
-    BOOST_STATIC_CONSTANT(bool, value = is_##name<                      \
-                               typename remove_cv<                      \
-                                  typename remove_reference<X_>::type   \
-                               >::type                                  \
-                           >::value);                                   \
-    typedef mpl::bool_<value> type;                                    \
-                                                                        \
+#  define BOOST_PYTHON_VALUE_IS_XXX_DEF(name, qualified_name, nargs)                           \
+template <class X_>                                                                            \
+struct value_is_##name                                                                         \
+{                                                                                              \
+    BOOST_PYTHON_IS_XXX_DEF(name,qualified_name,nargs)                                         \
+    BOOST_STATIC_CONSTANT(bool, value = is_##name<                                             \
+                               typename boost::python::detail::remove_cv<                      \
+                                  typename boost::python::detail::remove_reference<X_>::type   \
+                               >::type                                                         \
+                           >::value);                                                          \
+    typedef mpl::bool_<value> type;                                                            \
+                                                                                               \
 };                                                              
```

Comment:
It looks like right now (i.e. even before your change) the `value_is_...` template is defined in the global namespace. Can it be moved into boost::python::detail ?